### DISCUSSION TITLE
Percentage complete, activity progress

### DISCUSSION THREAD

**Discussion by JimmyRoasts** [kimai/kimai#4607] [General]
### Describe the problem

Hi Kevin, Love the project, just had a quick question to see how tricky this one is. Would it be possible to add a section that reports how much budget is remaning on acitivity? We're collecting % complete for each activity, so we can do an '[earned value](https://www.wrike.com/project-management-guide/faq/how-to-calculate-earned-value-in-project-management/)' calculation. We're having a little trouble knowing what hours remain on a given activity as we book them.

![image](https://github.com/kimai/kimai/assets/51733768/b3298b1f-4c61-44ed-ba58-d20b5da9bb09)

### Describe the solution you'd like

Ideally on the time entry form, if we could get a field called 'remaining budget' that shows activity budget - booked hours to that activity (search for timecards associated with acitivty, calculate total booked hours, minus from activity budget).  

### Describe alternatives you've considered

We've used an activity called "OVERFLOW" where people can put hours if they exceed and activity budget. The project manager can restribute the hours to appropriate activities. 

I've considered the custom content bundle too of a potential way to display the information, or perhaps custom CSS?

### Screenshots

Something like this is what I was thinking. This may be much too difficult or not possible at all. But if there is a possibilty of getting something similar working, please let me know. 

![image](https://github.com/kimai/kimai/assets/51733768/aabb39f7-a6f5-4dd8-afec-e2843b4a0a24)

**Comment by kevinpapst** [comment]
Interesting approach. 

Needs new API calls. Would need to check budget on customer, project and activity.
Then upon any selection change, this value needs to be reloaded via API and rendered in the form.
Pretty expensive calls calculation wise, as this budget status info is not cached anywhere yet.

I heard that wish before once or twice, there should be an existing issue, if I didn't close it.

**Reply by JimmyRoasts** [reply]
Cheers Kevin, I appreciate it. I've been having a think about this one too. If you scale this system up to say, 200 staff, filling it out on the same day, that could get computationally expensive. I think sharky is right too, this would be only really useful in a handful of cases. One might want to instead ask the question why are we using a timesheeting application to do project management. A plugin is outside my skillset unfortunately, but I was thinking of using a spreadsheet and power query to grab the data from kimai and display it that way. Have a dashboard / project management spreadsheet addin for excel and just grab the data from kimai. I think the main issue they were having was that overbook of stored budgets. I thought just have an overflow activity, and sort them out at the approval step. Thanks heaps guys for your thoughts.

**Comment by sharky98** [comment]
I think there is more than one issue related to that idea of the budget relation between customer, project and activity. #3584, #3306 and #3293 to name a few after a quick search. But I think all of them revolves around different views and concepts of a budget.

Talking for myself only, I don't think Kimai should try to cover all views and concepts regarding budget as this is not really the *core* objective of the tool. Budget options should limit themselves to the few options that would cover most "quick glance" use cases; leave the advanced use cases to external tool through API calls. However, Kevin may have something else in his mind; so what he has in mind will prevail 😉. Unless we convince him otherwise 😂.

**Reply by kevinpapst** [reply]
😁 mostly agree! People have many ideas, only a few are compatible with the pareto principle / 80% of all use-cases.
